controllers/pkg/reconcilers/bootstrap-packages/reconciler.go

```diff
@@ -89,144 +82,135 @@ func (r *reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, c a
 type reconciler struct {
 	client.Client
 	porchClient client.Client
-
-	l logr.Logger
+	log         logr.Logger
 }
 
 func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	r.l = log.FromContext(ctx)
-	cr := &porchv1alpha1.PackageRevision{}
-	if err := r.Get(ctx, req.NamespacedName, cr); err != nil {
+	r.log = log.FromContext(ctx)
+	packageRevision := &porchv1alpha1.PackageRevision{}
+	if err := r.Client.Get(ctx, req.NamespacedName, packageRevision); err != nil {
 		// There's no need to requeue if we no longer exist. Otherwise we'll be
 		// requeued implicitly because we return an error.
 		if resource.IgnoreNotFound(err) != nil {
 			msg := "cannot get resource"
-			r.l.Error(err, msg)
+			r.log.Error(err, msg)
 			return ctrl.Result{}, errors.Wrap(resource.IgnoreNotFound(err), msg)
 		}
 		return ctrl.Result{}, nil
 	}
 
 	// check if the packagerevision is part of a staging repository
 	// if not we can ignore this package revision
-	stagingPR, err := r.IsStagingPackageRevision(ctx, cr)
+	stagingPR, err := r.IsStagingPackageRevision(ctx, r.porchClient, packageRevision.Spec.RepositoryName)
 	if err != nil {
 		msg := "cannot list repositories"
-		r.l.Error(err, msg)
+		r.log.Error(err, msg)
 		return ctrl.Result{}, errors.Wrap(err, msg)
 	}
-	if stagingPR && porchv1alpha1.LifecycleIsPublished(cr.Spec.Lifecycle) {
-		r.l.Info("reconcile package revision")
-		resources, namespacePresent, err := r.getResources(ctx, req)
+	if stagingPR && porchv1alpha1.LifecycleIsPublished(packageRevision.Spec.Lifecycle) {
+		r.log.Info("reconcile package revision")
+		// get the relevant package revision resources
+		resources, err := r.getPrResources(ctx, req)
 		if err != nil {
 			msg := "cannot get resources"
-			r.l.Error(err, msg)
+			r.log.Error(err, msg)
 			return ctrl.Result{}, errors.Wrap(err, msg)
 		}
-		// we expect the clusterName to be applied to all resources in the
-		// package revision resources, so we find the cluster name by looking at the
-		// first resource in the resource list
 		if len(resources) > 0 {
-			clusterName, ok := resources[0].GetAnnotations()[clusterNameKey]
+			// we expect the clusterName to be applied to all resources in the
+			// package revision resources, so we find the cluster name by looking at the
+			// first resource in the resource list
+			clusterName, ok := resources[0].GetAnnotations()["nephio.org/cluster-name"]
 			if !ok {
-				r.l.Info("clusterName not found",
+				r.log.Info("clusterName not found",
 					"resource", fmt.Sprintf("%s.%s.%s", resources[0].GetAPIVersion(), resources[0].GetKind(), resources[0].GetName()),
 					"annotations", resources[0].GetAnnotations())
 				return ctrl.Result{}, nil
 			}
-			// we need to find the cluster client
-			secrets := &corev1.SecretList{}
-			if err := r.List(ctx, secrets); err != nil {
-				msg := "cannot list secrets"
-				r.l.Error(err, msg)
+			clusterSecret, err := r.GetClusterSecret(ctx, r.Client, clusterName)
```

Comment:
you should not have to put r.Client as a param

Reply:
Done